controlpanel/api/models.py

```diff
@@ -50,27 +53,51 @@ def aws_delete_role(self):
         services.delete_role(self.iam_role_name)
 
     def helm_create(self):
-        helm.init_user(self.username, self.email, self.get_full_name())
-        helm.config_user(self.username)
+        user_slug = sanitize_dns_label(self.username)
+        values = ",".join(f"{key}={value}" for key, value in {
+            "NFSHostname": settings.NFS_HOSTNAME,
+            "Username": user_slug,
+            "Email": self.email,
+            "Fullname": self.get_full_name(),
+            "Env": settings.ENV,
+            "OidcDomain": settings.OIDC_DOMAIN,
+        })
+        helm.upgrade_release(
+            f"init-user-{user_slug}",
+            "mojanalytics/init-user",
+            f"--set={values}",
+        )
+        helm.upgrade_release(
+            f"config-user-{user_slug}",
+            "mojanalytics/config-user",
+            "--namespace=user-{user_slug}",
+            "--set=Username={user_slug}",
+        )
 
     def helm_delete(self):
-        helm.uninstall_user_charts(self.username)
-        helm.uninstall_init_user_chart(self.username)
+        user_slug = sanitize_dns_label(self.username)
+        helm.delete(helm.list_releases(f"--namespace=user-{user_slug}"))
+        helm.delete(f"init-user-{user_slug}")
 
+    def is_app_admin(self, app_id):
+        return self.userapps.filter(app_id=app_id).count() != 0
 
-class App(TimeStampedModel):
-    def _slugify(name):
-        """Create a slug using standard django slugify but we override with
-        extra replacement so it's valid for s3"""
-        return re.sub(r'_+', '-', slugify(name))
+    def is_bucket_admin(self, bucket_id):
+        return self.users3buckets.filter(
+            s3bucket__id=bucket_id,
+            is_admin=True,
+        ).count() != 0
```

Comment:
#### 1b. Update models

> Lines 82 - 89

These new methods are used in the frontend templates

[Next →]()